audio/out/ao.c

```diff
@@ -102,12 +102,11 @@ static const struct ao_driver * const audio_out_drivers[] = {
 #endif
     &audio_out_pcm,
     &audio_out_lavc,
-    NULL
```

Comment:
`vo.c` has this too and works basically in the same way. Might be best to just keep it as it is.

Reply:
It seems much easier to work with the explicitly known length of the list.
Am I missing something?

I can also post a PR that changes both ao.c and vo.c to the length-based handling.

Reply:
Yeah I can't say I'm particularly a big fan of relying on NULL when we can just use MP_ARRAY_SIZE. A separate PR is fine imo.

Reply:
PR in #11148